### PR TITLE
Turn on prettier (except repo root)

### PR DESCRIPTION
**Pull request by paulmelnikow**
Follow-up to #1123.

This is going to conflict with open PRs. Before merging it, I want to get through a few open PRs which are making substantive changes in lib/:

- [x] badges/shields#1147
- [x] badges/shields#1142
- [x] badges/shields#1118

The open PRs for service badges will conflict too, though the conflicts there will be smaller and so not as difficult to solve.

### CODE REVIEW

**Comment by paulmelnikow** [comment]
I proposed at https://github.com/badges/shields/issues/948#issuecomment-337449221 that we turn off semicolons for lib/ and service-tests/ before merging this.

File: lib/badge.js [hunk 1]
```diff
@@ -31,9 +34,17 @@ templateFiles.forEach(function(filename) {
     });
     compressSvg(untemplatedSvg, function(object) {
       if (object.error !== undefined) {
-        console.error('Template ' + filename + ': ' + object.error + '\n' +
-          '  Generated untemplated SVG:\n' +
-          '---\n' + untemplatedSvg + '---\n');
+        console.error(
+          'Template ' +
```
Comment (Daniel15) [inline lib/badge.js]: Looks like a great use case for a format string.

**Comment by platan** [comment]
Do we want to add info about usage of prettier in this project in documentation, e.g. in `CONTRIBUTING.md`? And maybe one example command how to format code using `prettier` (or other tool).

**Comment by FezVrasta** [comment]
pre-commit hook would make much more sense than explaining commands to the contributors, the process should be transparent

**Comment by paulmelnikow** [comment]
Let's take one last pass through the recent open PRs that modify a lot of files in `lib/`, `frontend/`, services/`, and then rebase and merge this.

- [ ] #1762
- [ ] #1743
- [ ] #1740 
- [ ] #1725

**Comment by paulmelnikow** [comment]
I'm not sure we'll be able to merge this in a timely way. How about we just go ahead and do this, and deal with the rebasing + conflicts? I worry if we wait until everything is clear we could be waiting a long time…

**Comment by RedSparr0w** [comment]
Yeah, that sounds fine to me 👍

File: lib/badge.js (hunk 1 above)
Comment (RedSparr0w) [reply lib/badge.js]: Is there a prettier option for prefer template string?

**Comment by paulmelnikow** [comment]
@chris48s Does that work for you, too? Most of the outstanding rebasing work would likely fall on the three of us, so if we're cool with it, I'll revive this, update it, add some doc, and then merge it shortly after.

**Comment by paulmelnikow** [comment]
I spot-checked these and they look _way_ cleaner, especially without the semis. The tester code looks especially pretty! ✨

**Comment by Daniel15** [comment]
Nice work! I miss the semicolons, but consistency is good. I'm happy either way, as long as it's consistent across everything :)

**Comment by paulmelnikow** [comment]
To make it easier to read the config changes, I reverted the formatting changes, which I'll merge separately. This PR will fail `prettier-check` in CI.

**Comment by paulmelnikow** [comment]
I've posted guidance on resolving merge conflicts here: https://github.com/badges/shields/pull/1866#issuecomment-411567524

**Pull request merged**